orc8r/cloud/deploy/scripts/create_application_certs.sh

```diff
@@ -53,6 +53,13 @@ echo "######################"
 openssl genrsa -out fluentd.key 2048
 openssl req -x509 -new -nodes -key fluentd.key -sha256 -days 3650 -out fluentd.pem -subj "/C=US/CN=fluentd.$domain"
 
+echo ""
+echo "######################"
+echo "Creating nprobe certs"
+echo "######################"
+openssl genrsa -out nprobe.key 2048
+openssl req -x509 -new -nodes -key nprobe.key -sha256 -days 3650 -out nprobe.pem -subj "/C=US/CN=nprobe.$domain"
```

Comment:
Are the nprobe certs provided externally, or are they always generated? If they're always generated, we don't need to use a new keypair, we can just reuse the existing `controller.{key,crt}` keypair

Reply:
nprobe.{key, pem} are client certificates and will provided by LI vendors. 
During the setup, I generate these to allow the setup to go through.

Reply:
So, if the certs are going to be provided by operators, then that means, eventually, we'll want multi-tenant support

Which leads me to the idea of uploading these certs per-network via Swagger. Which would then require that we gracefully handle the certs not existing

So I think it would be cleaner to just not generate these certs (since, if I'm reading correctly, they're "throwaway" certs that will never be used ?). The nprobe codepath should handle the certs not existing

Let me know if I'm misunderstanding. My reason for pushing back on this is we're actively trying to *decrease* the number of certs we have to track, as cert issues is a perennial SEV rc. So actually adding new certs needs an air-tight rationale

Reply:
Moving the config including certs to swagger will not make the solution multi-tenant. As you know, a single admin_operator cert will provide access to all tenants data. Thus, we will not be able to expose X1 in multi-tenant environment anyway.

In addition, having a certificate with private key accessible through swagger without any access control due to current limitation seems to me a bigger security concern!

Anyway, your requested changes were added.

Reply:
Agreed that moving that part of the config to swagger doesn't achieve multi-tenant solution -- rather, pointing to the idea that, since we'll have to move it to swagger eventually (unless I'm misunderstanding), may as well do it now

Agreed that it's not ideal that a private key is accessible through swagger given our lack of fine-grained access controls. However, (a) that will be fixed on-demand (when partners request it we'll prioritize the fix), and (b) the current pattern of Orc8r usage is admin_operator is only ever used by NMS or admins who already have full access to the Orc8r infrastructure -- meaning moving the configs to swagger isn't any less secure under current patterns

Thanks for making the change 🙂